Source/Tools/FEXLoader/LinuxSyscalls/SignalDelegator.cpp

```diff
@@ -90,6 +90,63 @@ namespace FEX::HLE {
    * @name Signal frame setup
    * @{ */
 
+  // Total number of layouts that siginfo supports.
+  enum class SigInfoLayout {
+    LAYOUT_KILL,
+    LAYOUT_TIMER,
+    LAYOUT_POLL,
+    LAYOUT_FAULT,
+    LAYOUT_FAULT_RIP,
+    LAYOUT_CHLD,
+    LAYOUT_RT,
+    LAYOUT_SYS,
+  };
+
+  // Calculate the siginfo layout based on Signal and si_code.
+  static SigInfoLayout CalculateSigInfoLayout(int Signal, int si_code) {
+    SigInfoLayout Layout = SigInfoLayout::LAYOUT_KILL;
+    if (si_code > SI_USER && si_code < SI_KERNEL) {
+      // For signals that are not considered RT.
+      if (Signal == SIGSEGV ||
+          Signal == SIGBUS ||
+          Signal == SIGTRAP) {
+        // Regular FAULT layout.
+        Layout = SigInfoLayout::LAYOUT_FAULT;
```

Comment:
Wouldn't these also be able to be returns like in the else case below?

Reply:
oop, you're right. Fixed.